dask/utils.py

```diff
@@ -135,15 +141,29 @@ def filetexts(d, open=open):
             os.remove(filename)
 
 
-opens = {'gzip': gzip.open}
+compressions = {'gz': 'gzip', 'bz2': 'bz2'}
+
+
+def infer_compression(filename):
+    extension = os.path.splitext(filename)[-1].strip('.')
+    return compressions.get(extension, None)
+
+
+opens = {'gzip': GzipFile, 'bz2': BZ2File}
 
 
-def get_bom(fn):
+def open(filename, mode='rb', compression=None, **kwargs):
```

Comment:
You would have to make a class that implements `__init__`, `__enter__` and `__exit__`.  Some googling is probably required to do this well